### ISSUE TITLE
Date and time not retained when moving messages from one account to another

### ISSUE THREAD

**Issue by dburke555** [jstedfast/MailKit#1305]
Using the solution provided by you on Stackoverflow [here](https://stackoverflow.com/questions/51448444/mailkit-mail-client-is-showing-message-as-received-at-time-of-move-and-not-reta), the email messages are moving over, but the mail client is still showing their received date as the date/time it was moved (in the list of messages view), not the date/time that's showing in the message header.   I posted an answer in that Stackoverflow thread that also references this issue and included my code there.  I'm also including my code below.

```csharp
private void CopyEmailsAndFolders()
{
    try
    {  
        ImapClient yahooEmailAccnt = connectToEmailServerAndAccount(SrcEmailAccount.ImapServer, SrcEmailAccount.LogFileName, SrcEmailAccount.AccountUserName, SrcEmailAccount.AccountPassword);
        ImapClient gmailEmailAccnt = connectToEmailServerAndAccount(DestEmailAccount.ImapServer, DestEmailAccount.LogFileName, DestEmailAccount.AccountUserName, DestEmailAccount.AccountPassword);

        try
        {
            List<string> yahooFoldersList = getFoldersList(yahooEmailAccnt);                
            
            //loop through each of the folders in the source (Yahoo) email account
            foreach (string folderName in yahooFoldersList)
            {
                if (folderName.ToLower() == "inbox")
                {
                    continue;
                }

                var gmailFolder = gmailEmailAccnt.GetFolder(folderName);
                var yahooFolder = yahooEmailAccnt.GetFolder(folderName);

                yahooFolder.Open(FolderAccess.ReadOnly);

                //get a list of all email UID's
                var uids = yahooFolder.Search(SearchQuery.All);
                    
                //now get a list of all the items in the folder to include the critical date time properties we want to preserve
                var items = yahooFolder.Fetch(uids, MessageSummaryItems.InternalDate | MessageSummaryItems.Flags);

                //now loop through all of the emails we found in the current folder and copy them to the destination (gmail) email account
                foreach (var item in items)
                {
                    var message = yahooFolder.GetMessage(item.UniqueId);
                    gmailFolder.Append(message, item.Flags.Value, item.InternalDate.Value);
                }
            }
        }
        catch (Exception fEx)
        {
            MessageBox.Show(fEx.Message);
        }
        finally
        {
            yahooEmailAccnt.Disconnect(true);
            gmailEmailAccnt.Disconnect(true);
        }
    }
    catch (Exception ex)
    {
        MessageBox.Show(ex.Message);
    }
}

private ImapClient connectToEmailServerAndAccount(string imapServer, string logFile, string userName, string password)
{
    ImapClient newEmailClient = new ImapClient();
        
    try
    {
        newEmailClient = new ImapClient(new ProtocolLogger(logFile));
        newEmailClient.Connect(imapServer, 993, SecureSocketOptions.SslOnConnect);
        newEmailClient.Authenticate(userName, password);
    }
    catch (Exception ex)
    {
        MessageBox.Show(ex.Message);
    }

    return newEmailClient;
}

private List<string> getFoldersList(ImapClient emailClient)
{
    List<string> foldersList = new List<string>();
        
    try
    {
        // Get the first personal namespace and list the toplevel folders under it.
        var personal = emailClient.GetFolder(emailClient.PersonalNamespaces[0]);

        foreach (var folder in personal.GetSubfolders(false))
        {
            foldersList.Add(folder.Name);
            Console.WriteLine("[folder] {0}", folder.Name);
        }                
    }
    catch (Exception ex)
    {
        MessageBox.Show(ex.Message);
    }

    return foldersList;
}
```

**Comment by jstedfast** [comment]
If that doesn't work, there's nothing you or I can do about it. It's clearly a bug in the IMAP server that you are appending messages to.

**Comment by dburke555** [comment]
Just in case someone else runs into this in the future... I figured out what the issue was.  I had initially tried copying over a bunch of emails without using the flags that retains the date and time.  I didn't know that would be required.  So I then implemented the changes necessary to use those flags and before running the program again, I deleted all of the original emails that I had copied over from the previous version of my program, but I never emptied the trash in gmail.  So although I was copying over the messages again, it must have just used the original copied messages and just removed the deleted flag.  Once I emptied the gmail trash and ran my program again, the emails copied over correctly while retaining the original received date.
